### PR TITLE
[receiver/apache] update units to follow semconv

### LINKED ISSUE #10548: [apachereceiver] Some metric units don't follow Otel semantic conventions
**Describe the bug**
Some metrics don't follow [Otel semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md)

**Steps to reproduce**
Nothing to reproduce. The problem lies in the specification of the receiver itself.

**What did you expect to see?**
`apache.current_connections` unit must be `{connections}` (with curly brackets)
`apache.requests` unit must be `{requests}`

**What did you see instead?**
`apache.current_connections` with non-standard unit (and other similar metrics)
`apache.requests` with unit `1` (which means ratio, or fraction)

**What version did you use?**
Main branch

**What config did you use?**
N/A

**Environment**
N/A

**Additional context**
N/A

**Additional context**

### PR DESCRIPTION
**Pull request by codeboten**
Connections were already reported as connections, but addressed other units that didn't seem to follow conventions.

Fixes #10548

### CODE REVIEW

File: receiver/apachereceiver/metadata.yaml [hunk 1]
```diff
@@ -78,12 +78,12 @@ metrics:
...
   apache.scoreboard:
     enabled: true
-    description: The number of connections in each state.
+    description: The number of workers in each state.
     extended_documentation: >-
       The apache scoreboard is an encoded representation of the state of all the server's workers.
       This metric decodes the scoreboard and presents a count of workers in each state.
       Additional details can be found [here](https://support.cpanel.net/hc/en-us/articles/360052040234-Understanding-the-Apache-scoreboard).
-    unit: scoreboard
+    unit: workers
```
Comment (bertysentry) [inline receiver/apachereceiver/metadata.yaml:86]: `{workers}`

**Review by bertysentry — CHANGES_REQUESTED** [review]
Thank you for the update! Please surround non-standard units with `{}` as per [Otel's semantic conventions for metrics](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-units)

**Review by bertysentry — APPROVED** [review]

File: receiver/apachereceiver/metadata.yaml [hunk 2]
```diff
@@ -78,12 +78,12 @@ metrics:
     attributes: [ server_name ]
   apache.scoreboard:
     enabled: true
-    description: The number of connections in each state.
+    description: The number of workers in each state.
```
Comment (djaglowski) [inline receiver/apachereceiver/metadata.yaml:81]: I don't believe this change is accurate. My understanding is that the scoreboard represents workers, some of which may have active connections, but not all.

Comment (codeboten) [reply receiver/apachereceiver/metadata.yaml:81]: According to https://metacpan.org/pod/Apache::Scoreboard, it seems it represents server or workers:

> Apache keeps track of server activity in a structure known as the scoreboard . There is a slot in the scoreboard for each child server and its workers (be it threads or processes), containing information such as status, access count, bytes served and cpu time, and much more.

**Comment by codeboten** [comment]
@djaglowski please take another look

**Review by djaglowski — APPROVED** [review]

**Pull request merged**